### PR TITLE
Add support for GitHub Enterprise

### LINKED ISSUE #315: Add support for GitHub Enterprise domains
I think it's great that you've limited the extension to only accessing data on the github.com domain but this prevents me from using it with GitHub Enterprise which runs on an internal domain.

### PR DESCRIPTION
**Pull request by bfred-it**
Fixes #315 

New domains can be specified in GH's options view, which can also help #193, should that ever materialize.

options view

To test this without access to GHE:

1. remove all the permissions to `github.com` in `manifest.json`
42. disable and re-enable the extension (don't just reload it)
369. Verify the permissions, `github.com` should not be there
9001. Add `https://github.com` via the form

### CODE REVIEW

**Comment by RodEsp** [comment]
@bfred-it this is gold.
@sindresorhus could we get some eyes on this? I'd love to have it merged as soon as possible.

**Comment by busches** [comment]
Should there be an option to see currently added domains and remove them if no longer needed?

**Comment by bfred-it** [comment]
@busches I don't think it's a big deal, worst thing that can happen is that a domain you no longer use takes 20 bytes on disk, I think.

**Comment by bfred-it** [comment]
Alternative style:

**Comment by derimagia** [comment]
@bfred-it Very nice work! Tested this on GHE and the actual permission stuff worked great. All of the css changes worked without issue.

However the javascript doesn't fire at all because this is loading after the page loaded and the javascript changes are done using a listen on the DOMContentLoaded event and not taking into account if it's loaded after that event has triggered. Should be easy to fix.  content.js should probably be executed last (right now there are a few after it and just calling the content loaded function errors because of this)

It also looks like right now you're guaranteeing execution order - the javascript is noticeably slower to load on the GHE sites as well and this may be why. We should be able to load all of them async except for content.js and load that after, right?

**Comment by bfred-it** [comment]
Thanks for the info! I'll apply the two fixes (domready and content.js loaded last) but the speed will automatically be improved once #404 is in and only a file is loaded.

**Comment by paulmolluzzo** [comment]
I don't use GHE, so this PR is out of my reach I think. @hkdobrev maybe? Or @jgierer12?

**Comment by bfred-it** [comment]
Applied the 3 suggested changes and rebased.

**Comment by jgierer12** [comment]
Nope, I don't use GHE either

**Comment by derimagia** [comment]
@bfred-it Got it

Confirming this is working for me on GHE. Also note you can see what sites you gave permission to in the settings for the extension.

**Comment by bfred-it** [comment]
Thanks @derimagia! Hopefully this can be merged/squashed soon :)

File: extension/options/index.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
-<!DOCTYPE html>
-<meta charset="UTF-8">
+<!doctype html>
```
Comment (bfred-it) [inline extension/options/index.html]: It's case-insensitive, but the official one is actually uppercase: https://html.spec.whatwg.org/multipage/syntax.html#the-doctype

File: extension/options/index.html [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
+<meta charset="utf-8">
```
Comment (bfred-it) [inline extension/options/index.html]: This is fine 👍 

https://encoding.spec.whatwg.org/#concept-encoding-get

**Comment by busches** [comment]
This works well for the most part. There are still a lot of places that have `github.com` hardcoded, such as branch names, reaction avatars, show names, etc. These links do not work when you're on another domain as they incorrectly go to the wrong URL. This is a larger undertaking, so I can see it being moved out to another PR, but is core to supporting other domains.

**Comment by bfred-it** [comment]
@busches do you see value in merging this now to get at least some features working while further and specific issues can be reported and fixed later?

**Comment by busches** [comment]
@bfred-it yes. We'd just need to create an issue for what I mentioned above.

**Comment by bfred-it** [comment]
Cool. There's a chance that some features won't ever be available or compatible with GHE, but custom domain support at least gives a basis for other to find and fix issues too.

**Comment by derimagia** [comment]
@bfred-it I'd be happy to work with you as needed, what features aren't compatible with it would you say?

**Comment by busches** [comment]
@derimagia anything that has `github.com` hardcoded, instead of using something like `location.origin`:
* Linkify Branch References
* Add Delete Fork Link
* isGist function detection
* Reaction Avatar images
* Showing Real Name next to username
* Linkify issues

There could be more, but from a quick glance at the code and attempting them, they don't work.

**Comment by derimagia** [comment]
Sorry I mean't this part of it: "Some features won't ever be available or compatible with GHE" - is it that they can't be done or that we would need to fix the hard code locations?

**Comment by bfred-it** [comment]
I don't have GHE so I', only guessing, considering that GHE is not 1:1 with GH.com

**Comment by bfred-it** [comment]
The URLs probably only need to be changed from `https://github.com/something` to `/something`. But that can come in a different PR and further testing

**Comment by sindresorhus** [comment]
@busches Can you open an issue about further improvements to GHE support?

**Comment by sindresorhus** [comment]
@bfred-it Can you mention GHE support in the readme?

**Comment by jgierer12** [comment]
Just a heads-up, this doesn't work on Firefox since it doesn't allow changing permissions dynamically. I don't think there's much we can do about it currently, so maybe mention this in the readme too @bfred-it

**Comment by bfred-it** [comment]
Pinging some people who showed interest in Refined GitHub Enterprise:

* @nobso in #318
* @mwakerman, @thefotios in #315
* @zafarella in #81 

Would you mind giving it a go and reporting new issues?

**Comment by zafarella** [comment]
Thank you so much for adding the support. I was adding the GHE URL to manifest file as workaround.

Works for me but only after installing it from store. When loading locally, it does not work.
Extension was this red box
```
There were warnings when trying to install this extension:
Unrecognized manifest key 'applications'.
```
I pulled latest master and:
```
npm  install
npm run build
```

**Comment by bfred-it** [comment]
You don't need to change the manifest anymore, you can just set your domain in the options:

**Pull request merged**